### PR TITLE
 Refactor AgentPreOfferingPopUp & added AgentAfterOfferingPopUp

### PR DESCRIPTION
**Pull request by dzeckelev**
Resolves BV-716.

### CODE REVIEW

File: proc/worker/agent.go [hunk 1]
```diff
@@ -758,6 +759,62 @@ func (w *Worker) AgentPreOfferingDelete(job *data.Job) error {
...
 }
 
+func (w *Worker) agentOfferingPopUpFindRelatedJobs(
+	logger log.Logger, id string) error {
+	query := `SELECT count(*)
+		    FROM jobs
+		   WHERE (jobs.type = $1
+                         OR jobs.type = $2)
+                         AND jobs.status = $3
+		         AND jobs.related_id = $4;`
```
Comment (ababo) [inline proc/worker/agent.go]: Fix the indent please.

File: proc/worker/agent_test.go [hunk 2]
```diff
@@ -660,6 +660,34 @@ func TestAgentPreOfferingPopUp(t *testing.T) {
 	fxt.Offering.OfferStatus = data.OfferRegister
 	env.updateInTestDB(t, fxt.Offering)
 
+	if err := env.worker.AgentPreOfferingPopUp(
+		fxt.job); err != ErrUncompletedJobsExists {
+		t.Fatal("active jobs not exists")
```
Comment (ababo) [inline proc/worker/agent_test.go]: no active jobs

File: proc/worker/agent_test.go [hunk 3]
```diff
@@ -671,3 +699,39 @@ func TestAgentPreOfferingPopUp(t *testing.T) {
...
+
+	fxt.job.RelatedID = util.NewUUID()
+	env.updateInTestDB(t, fxt.job)
+
+	offeringHash := data.TestToHash(t, fxt.Offering.Hash)
+	agentAddr := data.TestToAddress(t, fxt.Account.EthAddr)
+
+	topics := data.LogTopics{
+		common.BytesToHash(agentAddr.Bytes()),
+		common.Hash{},
```
Comment (ababo) [inline proc/worker/agent_test.go]: Doesn't make much sense, please prefer designated initialisers instead.

File: proc/worker/agent.go (hunk 1 above)
Comment (dzeckelev) [reply proc/worker/agent.go]: In the source everything is correct. This is the vision of the Githab.

Comment (ababo) [reply proc/worker/agent.go]: Very unlikely.

**Pull request merged**